src/controllers/auth.ts

```diff
@@ -0,0 +1,168 @@
+import { Request, Response } from 'express';
+import { IUser, User } from '../models/user.model';
+import { compare, genSalt, hash } from 'bcryptjs';
+import { Secret, sign, verify } from 'jsonwebtoken';
+
+export const register = async (req: Request, res: Response) => {
+  const { email, password, username } = req.body;
+
+  if (!email || !password || !username) {
+    res.json({ error: 'Please fill all fields' });
+    return;
+  }
+
+  try {
+    const user = await User.findOne({ email });
+    if (user) {
+      res.json({ error: 'User already exists' });
+      return;
+    }
+
+    const salt = await genSalt(10);
+    const encryptedPassword = await hash(password, salt);
+    const newUser: IUser = new User({
+      email,
+      password: encryptedPassword,
+      username,
+    });
+    await newUser.save();
+
+    res.json({ message: 'User registered successfully', newUser });
+  } catch (error: any) {
+    console.error('Error registering user: ', error.message);
+    res.status(403).json({ error: error.message });
+  }
+};
+
+export const login = async (req: Request, res: Response) => {
+  const { username, password } = req.body;
+
+  if (!username || !password) {
+    res.json({ error: 'Please fill all fields' });
+    return;
+  }
+
+  try {
+    const user = await User.findOne({ username });
+    if (!user) {
+      res.json({ error: 'User does not exist' });
+      return;
+    }
+
+    const isMatch = await compare(password, user.password);
```

Comment:
its return user.password without the salt? 
because in the register you enter with salt